### PR TITLE
[docs] documentation for 2.2 - read-only queries - cleos updates

### PR DESCRIPTION
**Pull request by halsaphi**
## Change Description

documentation only

## Change Type
**Select *ONE*:**
- [ x] Documentation

- [ ] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

This PR is documentation chnages required for read-only queries.

### CODE REVIEW

File: docs/02_cleos/03_command-reference/push/push-transaction.md [hunk 1]
```diff
@@ -1,36 +1,163 @@
...
-`--max-net-usage` _UINT_ - set an upper limit on the net usage budget, in bytes, for the transaction (defaults to 0 which means no limit)
+1. Basic usage
+```shell
+cleos push transaction {}
+```
 
-`--delay-sec` _UINT_ - set the delay_sec seconds, defaults to 0s
+2. Call a read-only query
+```shell
+cleos push transaction --read-only -j '{"actions":[{"account":"scholder","name":"get","authorization":[{"actor":"bob","permission":"active"}],"data":""}]}' -p bob@active
```
Comment (jgiszczak) [inline docs/02_cleos/03_command-reference/push/push-transaction.md:53]: Target account and permission again don't match, which is not usual for simple examples.

**Review by jgiszczak — CHANGES_REQUESTED** [review]

File: docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    ],
+    "account_ram_delta": null,
+    "except": null,
+    "error_code": null
+  }
+}
+```
+
+## Summary
+In conclusion, by following these instructions you are call a smart contract read-only action.
```
[the remark this answers is not in the document]
Comment (halsaphi) [reply docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md:113]: it's a template, though I agree it is awkward. Anyway those are the words we use.

[the remark this answers is not in the document]
Comment (lparisc) [reply docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md:113]: you are call --> you are able to call

File: docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+## Overview
+This guide provides instructions on calling a smart contract read-only query.   
+
+The example uses `cleos push transaction` to call a read-only query.  A readonly query can be called using either the `--read-only` option or the ``-o` option.
```
[the remark this answers is not in the document]
Comment (halsaphi) [reply docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md:4]: fixed

File: docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+## Before you Begin
+Make sure you meet the following requirements: 
+
+* Install version 2.2, or greater, of `cleos`  
+[[info | Note]]
+| `Cleos` is bundled with the EOSIO software. [Installing EOSIO](../../00_install/index.md) will also install the `cleos` and `keosd` comand line tools. 
+* You have loaded a smart contract, containing a read-only query to an EOSIO blockchain see 
+
+## Command Reference
```
[the remark this answers is not in the document]
Comment (halsaphi) [reply docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md:14]: no -> we have a "standard" template for these

File: docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+* [cleos push transaction](../03_command-reference/push/push-transaction.md) command
+
+## Examples
+
+The following examples show you how to call read-only queries:
+
+Example 1. Run the following command to call a read-only query _get_, with no parameters, on the _scholder_ account to which a smart contract containing a read-only query is deployed. The transaction is authorised by _bob@active_
+
+```shell
+cleos push transaction --read-only -j '{"actions":[{"account":"scholder","name":"get","authorization":[{"actor":"bob","permission":"active"}],"data":""}]}' -p bob@active
```
[the remark this answers is not in the document]
Comment (halsaphi) [reply docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md:26]: this form shows that other accounts can call the action and is widely used in the documentation

File: docs/02_cleos/03_command-reference/push/push-transaction.md (hunk 1 above)
Comment (halsaphi) [reply docs/02_cleos/03_command-reference/push/push-transaction.md:53]: I've updated the where explanation to make this clearer - but again don't see why you would use the same account as the smart contract.

File: docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+```
+
+**Where**
```
[the remark this answers is not in the document]
Comment (halsaphi) [reply docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md:29]: removed

File: docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (halsaphi) [reply docs/02_cleos/02_how-to-guides/how-to-call-smart-contract-read-only-queries.md:113]: fixed this

**Review by jgiszczak — APPROVED** [review]

**Pull request merged**